src/core/Akka.Remote/Serialization/ActorRefResolveCache.cs

```diff
@@ -65,7 +65,8 @@ protected override int Hash(string k)
 
         protected override bool IsCacheable(IActorRef v)
         {
-            return !(v is EmptyLocalActorRef);
+            // don't cache any FutureActorRefs, et al
+            return !(v is MinimalActorRef && !(v is FunctionRef));
```

Comment:
Filter out any temp actors from being added to the cache unless they're a `FunctionRef` - which is a rare but necessary edge case to support some Akka.Streams scenarios.

Should resolve https://github.com/akkadotnet/akka.net/issues/5230